packages/module/src/Status/Status.tsx

```diff
@@ -0,0 +1,92 @@
+import * as React from 'react';
+import clsx from 'clsx';
+import { Button, ButtonVariant, Flex, FlexItem, Popover, PopoverPosition, PopoverProps, Text, TextVariants, } from '@patternfly/react-core';
+import { createUseStyles } from 'react-jss';
+
+export const StatusVariant = {
+  link: 'link',
+  popover: 'popover',
+  plain: 'plain'
+} as const;
+
+export type StatusVariant = typeof StatusVariant[keyof typeof StatusVariant];
+
+export interface StatusProps extends React.PropsWithChildren {
+  /** Status label text */
+  label?: string;
+  /** Description to be displayed under the label */
+  description?: string;
+  /** If true, only displays icon */
+  iconOnly?: boolean;
+  /** Variant of the status component to be displayed */
+  variant?: StatusVariant;
+  /** Status icon */
+  icon?: React.ReactElement;
+  /** Custom OUIA ID */
+  ouiaId?: string | number;
+  /** Props for the optional popover */
+  popoverProps?: PopoverProps,
+  /** Optional link variant onClick callback */
+  onClick?: (event: React.MouseEvent<HTMLElement, MouseEvent>) => void;
+}
+
+const useStyles = createUseStyles({
+  icon: {
+    margin: 0
+  },
+  statusLabel: {
+    lineHeight: 'var(--pf-v5-global--LineHeight--sm)',
+  },
+  statusDescription: {
+    color: 'var(--pf-v5-c-content--small--Color)',
+  } 
+})
+
+export const Status: React.FC<StatusProps> = ({ variant = StatusVariant.plain, label, children, iconOnly, icon, ouiaId = 'Status', popoverProps, onClick, description, ...props }: StatusProps) => {
+  const classes = useStyles();
+
+  const statusBody = (
+    <Flex title={label} alignItems={{ default: 'alignItemsCenter' }} {...props}>
+      { icon && (
+        <FlexItem className={classes.icon}>
+          {React.cloneElement(icon, { className: clsx('pf-v5-u-mr-md', icon?.props?.className), title: label, 'data-ouia-component-id': `${ouiaId}-icon` })}
```

Comment:
By reusing the label as the icon's title, it causes the screen reader to repeat itself when it reads the icon and then reads the label. 

And since the label is marked as optional, when people don't want to display a label, they may leave it blank and leave the icon with no title - thus leaving it inaccessible. I wonder if the icon's title should be an additional field which defaults to the value of the `status` passed to the `Icon` component? and the status should be a required field.

Then it _should_ be overridden and given a more descriptive title when using the `iconOnly` prop. Maybe we could print a message in the console if someone uses `iconOnly` and then doesn't provide a descriptive Icon title.